lib/ledger_sync/adaptors/mixins/offset_and_limit_pagination_searcher_mixin.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module LedgerSync
+  module Adaptors
+    module Mixins
+      module OffsetAndLimitPaginationSearcherMixin
+        def next_searcher
+          paginate(limit: limit, offset: offset + limit)
+        end
+
+        def previous_searcher
+          return nil if offset <= 1
+
+          paginate(limit: limit, offset: offset - limit)
+        end
+
+        private
+
+        def default_offset
+          0
+        end
```

Comment:
Does NS start at 0?

It may be better to just raise in here (and `default_limit`) and force the class that includes the mixin to define it. I'm not sure if one value will truly be a default. So it'd be more like an interface